lib/chef/resource/aws_security_group.rb

```diff
@@ -1,14 +1,23 @@
-require 'chef/provisioning/aws_driver/aws_resource_with_entry'
+require 'chef/provisioning/aws_driver/aws_resource'
 require 'chef/resource/aws_vpc'
 require 'chef/provisioning/aws_driver/exceptions'
 
 class Chef::Resource::AwsSecurityGroup < Chef::Provisioning::AWSDriver::AWSResource
-  aws_sdk_type AWS::EC2::SecurityGroup, id: :name
+  aws_sdk_type AWS::EC2::SecurityGroup,
+               id: :id,
+               option_names: [:security_group, :security_group_id, :security_group_name]
 
   attribute :name,          kind_of: String, name_attribute: true
   attribute :vpc,           kind_of: [ String, AwsVpc, AWS::EC2::VPC ]
   attribute :description,   kind_of: String
 
+  # This should be a hash of tags to apply to the AWS object
+  # TODO this is duplicated from AWSResourceWithEntry
+  #
+  # @param aws_tags [Hash] Should be a hash of keys & values to add.  Keys and values
+  #        can be provided as symbols or strings, but will be stored in AWS as strings.
+  attribute :aws_tags, kind_of: Hash
```

Comment:
Is this how we really want to handle this?  I was hoping to avoid adding arbitrary aws_tag attributes.